### PR TITLE
Adapted to class structure style policy - Part 2

### PR DESCRIPTION
**Pull request by thomasfanell**
This PR covers:
* source/draw

### CODE REVIEW

File: source/draw/Color.ooc [hunk 1]
```diff
@@ -256,9 +248,23 @@ ColorBgra: cover {
...
...6468, -16110, -15752, -15394, -15036, -14678, -14320, -13962, -13604, -13246, -12888, -12530, -12172, -11814, -11456, -11098, -10740, -10382, -10024, -9666, -9308, -8950, -8592, -8234, -7876, -7518, -7160, -6802, -6444, -6086, -5728, -5370, -5012, -4654, -4296, -3938, -3580, -3222, -2864, -2506, -2148, -1790, -1432, -1074, -716, -358, 0, 358, 716, 1074, 1432, 1790, 2148, 2506, 2864, 3222, 3580, 3938, 4296, 4654, 5012, 5370, 5728, 6086, 6444, 6802, 7160, 7518, 7876, 8234, 8592, 8950, 9308, 9666, 10024, 10382, 10740, 11098, 11456, 11814, 12172, 12530, 12888, 13246, 13604, 13962, 14320, 14678, 15036, 15394, 15752, 16110, 16468, 16826, 17184, 17542, 17900, 18258, 18616, 18974, 19332, 19690, 20048, 20406, 20764, 21122, 21480, 21838, 22196, 22554, 22912, 23270, 23628, 23986, 24344, 24702, 25060, 25418, 25776, 26134, 26492, 26850, 27208, 27566, 27924, 28282, 28640, 28998, 29356, 29714, 30072, 30430, 30788, 31146, 31504, 31862, 32220, 32578, 32936, 33294, 33652, 34010, 34368, 34726, 35084, 35442, 35800, 36158, 36516, 36874, 37232, 37590, 37948, 38306, 38664, 39022, 39380, 39738, 40096, 40454, 40812, 41170, 41528, 41886, 42244, 42602, 42960, 43318, 43676, 44034, 44392, 44750, 45108, 45466 ]
```
Comment (marcusnaslund) [inline source/draw/Color.ooc]: I think there was a very good point to having this at the very end of the file, style guides or not.

File: source/draw/RasterImage.ooc [hunk 2]
```diff
@@ -57,6 +57,13 @@ RasterImage: abstract class extends Image {
 //		TODO: The stuff
 		result
 	}
+	save: virtual func (filename: String) -> Int { Debug raise("RasterImage save unimplemented for format!"); 0 }
+	kean_draw_rasterImage_getStride: unmangled func -> UInt { this stride }
+	kean_draw_rasterImage_save: unmangled func (path: const Char*) {
+		pathString := String new(path)
+		this save(pathString)
+		pathString free()
+	}
```
Comment (marcusnaslund) [inline source/draw/RasterImage.ooc]: Unmangled functions at the end of the class, we said. Even below static methods? I think so. You disagree?

File: source/draw/gpu/GpuSurface.ooc [hunk 3]
```diff
@@ -28,6 +28,11 @@ GpuSurface: abstract class extends Canvas {
 	_view := FloatTransform3D identity
 	_projection: FloatTransform3D
 	_toLocal := FloatTransform3D createScaling(1.0f, -1.0f, -1.0f)
+	_focalLength: Float
+	_nearPlane := 1.0f
+	_farPlane := 10000.0f
+	_defaultMap: GpuMap
+	_coordinateTransform := IntTransform2D identity
```
Comment (marcusnaslund) [inline source/draw/gpu/GpuSurface.ooc]: Yeah, this looks so much better.

File: source/draw/Color.ooc (hunk 1 above)
Comment (thomasfanell) [inline source/draw/Color.ooc]: Haha, are we going to live by this guide or not? How would we know when to deviate?

File: source/draw/RasterImage.ooc (hunk 2 above)
Comment (thomasfanell) [inline source/draw/RasterImage.ooc]: My reasoning is:
* methods
* unmangled methods
* static methods
* unmangled methods

Comment (marcusnaslund) [inline source/draw/RasterImage.ooc]: Didn't realize we also had static unmangled functions until further down. Agreed.

File: source/draw/Color.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/draw/Color.ooc]: To quote Star Trek: "There can be no justice so long as laws are absolute".

The first main rule says `Optimize for readability` and this seriously impacts that.

Comment (thomasfanell) [inline source/draw/Color.ooc]: Fair enough

**Comment by marcusnaslund** [comment]
Note that magic fails in two places.

File: source/draw/Color.ooc (hunk 1 above)
Comment (thomasfanell) [inline source/draw/Color.ooc]: Fixed

**Pull request merged**